DemoProject/DemoProject/ReverseKT/GolfHomeViewController.swift

```diff
@@ -0,0 +1,271 @@
+                return UITableViewCell()
+            }
+            cell.configureCell(data: groupedPlayerData[indexPath.section], sectionNumber: indexPath.section)
+            cell.chatButtonClickedClosure  = { [weak self] isSelected in
+                self?.groupedPlayerData[indexPath.section].isSectionIconClick?.toggle()
+                for i in 0...(self?.groupedPlayerData[indexPath.section].playerData.count ?? 1) - 1  {
+                    if ((self?.groupedPlayerData[indexPath.section].playerData[i].isChatVisible) != nil) {
+                        self?.groupedPlayerData[indexPath.section].playerData[i].isChatClicked = isSelected
+                    }
+                }
+                tableView.reloadSections([indexPath.section], with: .none)
+            }
+            
+            return cell
+        } else if indexPath.row == 1 {
+            print("for row at Section \(indexPath.section) Row \(indexPath.row)")
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: "GroupTableViewCell", for: indexPath) as? GroupTableViewCell else {
+                return UITableViewCell()
+            }
+            
+            cell.configureCell(data: nil, row: indexPath.row)
+            return cell
+        } else {
+            print("for row at Section \(indexPath.section) Row \(indexPath.row)")
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: "GroupTableViewCell", for: indexPath) as? GroupTableViewCell else {
+                return UITableViewCell()
+            }
+          
+            cell.changeButton = {
+                [weak self]
+                isSelected in
+                self?.groupedPlayerData[indexPath.section].playerData[indexPath.row - 2].isChatClicked?.toggle()
+                let clickedItemcount = self?.groupedPlayerData[indexPath.section].playerData.filter { $0.isChatClicked == true }.count
+                let totalItemCount = self?.groupedPlayerData[indexPath.section].playerData.filter { $0.isChatVisible == true }.count
+                self?.groupedPlayerData[indexPath.section].isSectionIconClick =  clickedItemcount == totalItemCount
+                tableView.reloadSections([indexPath.section], with: .none)
+            }
+
+            cell.configureCell(data: groupedPlayerData[indexPath.section].playerData[indexPath.row - 2] , row: indexPath.row)
+            return cell
+        }
+    }
+    
+}
+
+//MARK: - Tableview Delegate
+extension GolfHomeViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+    
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        print("Section \(indexPath.section) Row \(indexPath.row)")
```

Comment:
remove print statement if not needed

Reply:
that print is to check for which cell this method is called, i have added comment there